src/lib/Microsoft.Health.Extensions.Fhir.R4/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Health.Extensions.Host.Auth;
+using Microsoft.Health.Logging.Telemetry;
+using FhirClient = Microsoft.Health.Fhir.Client.FhirClient;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void AddFhirClient(this IServiceCollection serviceCollection, IConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(serviceCollection, nameof(serviceCollection));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            var url = new Uri(configuration.GetValue<string>("FhirService:Url"));
+            bool useManagedIdentity = configuration.GetValue<bool>("FhirClient:UseManagedIdentity");
+
+            serviceCollection.AddSingleton(typeof(ITelemetryLogger), typeof(IomtTelemetryLogger));
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            var logger = serviceProvider.GetRequiredService<ITelemetryLogger>();
+
+            serviceCollection.AddHttpClient<IFhirClient, FhirClient>(client =>
+            {
+                client.BaseAddress = url;
+                client.Timeout = TimeSpan.FromSeconds(60);
+
+                // Using discard because we don't need result
+                var fhirClient = new FhirClient(client);
+                _ = fhirClient.ValidateFhirClientAsync(logger);
```

Comment:
>_ = fhirClient.ValidateFhirClientAsync(logger); [](http://example.com/codeflow?start=16&length=47)

If we aren't waiting, is there value to doing this call when the class is created?

Reply:
From my perspective, the value in doing this call when the class is created is that the ValidateFhirClientAsync() executes right after the services are configured and built. This surfaces any invalid FhirClient configurations to the user before ingesting events from the input event hub

In addition by calling the validation method here, we don't need to call it across various classes for the console project and function apps